client/src/components/breadcrumbs.jsx

```diff
@@ -0,0 +1,37 @@
+//import { useState, useEffect } from 'react'
+//import PropTypes from 'prop-types'
+import './Breadcrumbs.css'
+
+// internal logic for breadcrumbs:
+// "/"   -->  ["home"]
+// "/one" --> ["home", "one"]
+// "/one/one_one" --> ["home", "one", "one_one"] --> home / one / one_one
+//
+// hint:
+// ["home", "one", "one_one"].join("/")
+
+const Breadcrumbs = ({ currentPath, fetchFiles }) => {
+  const pathnames = currentPath === '/' ? [] : currentPath.split('/').slice(1)
+
+  return (
+    <nav aria-label='Breadcrumb' className='breadcrumbs'>
+      <ul className='breadcrumb'>
+        <li className='breadcrumb-item' onClick={() => fetchFiles('/')}>
+          home
+        </li>
+        {currentPath !== '/' &&
+          pathnames.map((name, index) => {
+            const path = `/${pathnames.slice(0, index + 1).join('/')}`

```

Comment:
no need to use `pathnames`, 
better utilize third parameter for the callback that you pass into the map() method
refer to the documentation here: 
[Array/map#parameters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map#parameters)